### ISSUE TITLE
Question about scripting conditional event

### ISSUE THREAD

**Issue by ErogeSeiyuu** [Masuzu/ZooeyBot#489]
Hello, I don't know how to script this scenario:

1. Boss enter OD state
2. s.bea use her 1 twice to get maximum DATA, use her 2 for ODA.
3. auto attack
4. The following turn s.bea use her 1 again to remove def-down effect.

Currently, my script is as follow:
`   if turn >= 2 and enemy_1.overdrive_state == true then  `
`	characters["Beatrix"]:UseSkill(1)  `
`	characters["Beatrix"]:UseSkill(1)  `
`	characters["Beatrix"]:UseSkill(2)  `
`   end  `
`   if enemy_1.overdrive_state == true and characters["Beatrix"].skill_2_available == false then  `
`	characters["Beatrix"]:UseSkill(1)  `
`   end`

it would make sbea immediately use her 1st skill after using her 2nd skill.

how do I fix this?

**Comment by calvinhyang** [comment]
Reversing the order should (kind of) work. You also need to add additional conditionals otherwise you're just going to use Beatrix's 1 3 times a turn as long as the boss is in OD which I doubt is what you want. 

if enemy_1.overdrive_state == true and characters["Beatrix"].skill_2_available == false then
characters["Beatrix"]:UseSkill(1)
end
if turn >= 2 and enemy_1.overdrive_state == true and characters["Beatrix"].skill_2_available then
characters["Beatrix"]:UseSkill(1)
characters["Beatrix"]:UseSkill(1)
characters["Beatrix"]:UseSkill(2)
end

Problem with this is that you're still going to be using Beatrix's one once a turn if it's in OD + beatrix 2 is down. Easiest fix is just to hard code the turn that you want to turn off Beatrix's one, otherwise you probably have to start getting fancy with booleans. (like setting a boolean to false on turn 1, setting the boolean to true when you enter the second if conditional, and setting it back to false when you enter the first if conditional)

**Comment by ErogeSeiyuu** [comment]
Do you know a way to find the different status ID? Documentations does not seem to explain how to do it. 

I want to make sure Sturm use her 1 whenever she has her ougi passive available, but it's not listed in the documentation. Same goes to similar characters like Perci, Lancelot, Izmir/s.Izmir etc.

**Comment by calvinhyang** [comment]
If you right click on the buff icon (The icons in the In effect panel) and inspect element, you,'ll be lead to a link giving the source URL of the buff icon which conveniently has the status number in its link. 

For example, Beatrix when she uses her 1 twice. 
http://game-a1.granbluefantasy.jp/assets_en/img_mid/sp/ui/icon/status/x64/status_16451.png
So you can actually fix your earlier problem by just adding an if conditional that checks if she has this status up. (Assuming you want to turn it off the turn after you use her ODA combo)

**Comment by Hedova928** [comment]
Here is what I use. The key is do the deactivation check first and the activation check second, so the deactivation doesn't get triggered until the turn after the activation.

```lua
if characters["Beatrix"] ~= nil then
    -- If skill 2 is on CD and skill 1 has been activated twice, turn it off by activating it once
    if not characters["Beatrix"].skill_2_available and (characters["Beatrix"]:HasStatusEffect("16451")) then
        characters["Beatrix"]:UseSkill(1)
    end 
    -- If skill 2 is on CD and skill 1 has been activated once, turn it off by activating it twice
    if not characters["Beatrix"].skill_2_available and (characters["Beatrix"]:HasStatusEffect("1645")) then
        characters["Beatrix"]:UseSkill(1)
        characters["Beatrix"]:UseSkill(1)
    end 
    -- If enemy is in overdrive and ODA is available, activate skill 1 twice and use ODA
    if enemy_1.overdrive_state == true and characters["Beatrix"].skill_2_available then
            characters["Beatrix"]:UseSkill(1)
            characters["Beatrix"]:UseSkill(1)
            characters["Beatrix"]:UseSkill(2)
    end
    -- Use pots to keep her HP up so she can use her skill 3
    if booleans["greenPotUsed_2"] == true and characters["Beatrix"].hp_percentage < 50 then
        UseBluePotion()
    end
    if characters["Beatrix"].hp_percentage < 50 then
        UseGreenPotionOnPartyMember(4)
        if booleans["greenPotUsed_1"] == nil then
            booleans["greenPotUsed_1"] = true
        else
            booleans["greenPotUsed_2"] = true
        end
    end
    if characters["Beatrix"].hp_percentage >= 50 and not characters["Beatrix"]:HasStatusEffect("1181") then
        characters["Beatrix"]:UseSkill(3)
    end
end
```

**Comment by Masuzu** [comment]
Hello,
I will add a much more convenient feature the next week to capture the IDs of status effects of interest, taking the form of a new "listener" mode.

**Comment by Masuzu** [comment]
As announced, a new "Listener mode" is now available as of v2.6.0
<blockquote class="twitter-tweet" data-lang="en"><p lang="en" dir="ltr"><a href="https://twitter.com/hashtag/GranblueFantasy?src=hash&amp;ref_src=twsrc%5Etfw">#GranblueFantasy</a> <a href="https://twitter.com/hashtag/ZooeyBot?src=hash&amp;ref_src=twsrc%5Etfw">#ZooeyBot</a> A new listener mode is available as of version 2.6.0 to conveniently retrieve status effect IDs and use them in your Lua scripts. This will prove useful for the upcoming GW to automate soloing the NM 95 and 100 modes! <a href="https://t.co/mziLICB2Kl">https://t.co/mziLICB2Kl</a></p>&mdash; GBTools (@GBFPokerBot) <a href="https://twitter.com/GBFPokerBot/status/953048587390332928?ref_src=twsrc%5Etfw">January 15, 2018</a></blockquote>
